remote.go

```diff
@@ -0,0 +1,451 @@
+	}
+
+	// Build cache data structure
+	cache := &remoteCacheData{
+		evmChainIdToChainSelector:    data.Evm,
+		evmChainsBySelector:          make(map[uint64]Chain),
+		evmChainsByEvmChainID:        make(map[uint64]Chain),
+		solanaChainIdToChainSelector: data.Solana,
+		solanaChainsBySelector:       make(map[uint64]SolanaChain),
+		aptosSelectorsMap:            data.Aptos,
+		aptosChainsBySelector:        make(map[uint64]AptosChain),
+		suiSelectorsMap:              data.Sui,
+		suiChainsBySelector:          make(map[uint64]SuiChain),
+		tonSelectorsMap:              data.Ton,
+		tonChainIdBySelector:         make(map[uint64]int32),
+		tronSelectorsMap:             data.Tron,
+		tronChainIdBySelector:        make(map[uint64]uint64),
+		starknetSelectorsMap:         data.Starknet,
+		starknetChainsBySelector:     make(map[uint64]StarknetChain),
+		fetchedAt:                    time.Now(),
+	}
+
+	// Build EVM lookup maps
+	for chainID, details := range data.Evm {
+		chain := Chain{
+			EvmChainID: chainID,
+			Selector:   details.ChainSelector,
+			Name:       details.ChainName,
+		}
+		cache.evmChainsBySelector[details.ChainSelector] = chain
+		cache.evmChainsByEvmChainID[chainID] = chain
+	}
+
+	// Build Solana lookup maps
+	for chainID, details := range data.Solana {
+		chain := SolanaChain{
+			ChainID:  chainID,
+			Selector: details.ChainSelector,
+			Name:     details.ChainName,
+		}
+		cache.solanaChainsBySelector[details.ChainSelector] = chain
+	}
+
+	// Build Aptos lookup maps
+	for chainID, details := range data.Aptos {
+		chain := AptosChain{
+			ChainID:  chainID,
+			Selector: details.ChainSelector,
+			Name:     details.ChainName,
+		}
+		cache.aptosChainsBySelector[details.ChainSelector] = chain
+	}
+
+	// Build Sui lookup maps
+	for chainID, details := range data.Sui {
+		chain := SuiChain{
+			ChainID:  chainID,
+			Selector: details.ChainSelector,
+			Name:     details.ChainName,
+		}
+		cache.suiChainsBySelector[details.ChainSelector] = chain
+	}
+
+	// Build Ton lookup maps
+	for chainID, details := range data.Ton {
+		cache.tonChainIdBySelector[details.ChainSelector] = chainID
+	}
+
+	// Build Tron lookup maps
+	for chainID, details := range data.Tron {
+		cache.tronChainIdBySelector[details.ChainSelector] = chainID
+	}
+
+	// Build Starknet lookup maps
+	for chainID, details := range data.Starknet {
+		chain := StarknetChain{
+			ChainID:  chainID,
+			Selector: details.ChainSelector,
+			Name:     details.ChainName,
+		}
+		cache.starknetChainsBySelector[details.ChainSelector] = chain
+	}
+
+	// Update cache if TTL is set
+	if config.CacheTTL > 0 {
+		remoteCacheLock.Lock()
+		remoteCache = cache
+		remoteCacheLock.Unlock()
+	}
+
+	return cache, nil
+}
+
+// applyOptions applies functional options and returns a config with defaults
+func applyOptions(opts []RemoteOption) *RemoteConfig {
+	config := &RemoteConfig{
+		URL:      DefaultGitHubRawURL,
+		Timeout:  DefaultRemoteFetchTimeout,
+		CacheTTL: 5 * time.Minute,
```

Comment:
minor nit: Can we add this to const above, easier for people to read & understand